packages/aws-cdk/lib/api/evaluate-cloudformation-template.ts

```diff
@@ -74,22 +81,34 @@ export class EvaluateCloudFormationTemplate {
     this.region = props.region;
     this.partition = props.partition;
     this.urlSuffix = props.urlSuffix;
+    this.sdk = props.sdk;
+
+    // We need names of nested stack so we can evaluate cross stack references
+    this.nestedStackNames = props.nestedStackNames ?? {};
+
+    // The current resources of the Stack.
+    // We need them to figure out the physical name of a resource in case it wasn't specified by the user.
+    // We fetch it lazily, to save a service call, in case all hotswapped resources have their physical names set.
+    this.stackResources = new LazyListStackResources(this.sdk, this.stackName);
```

Comment:
why create a new one here, and have to pass the SDK through? Why do we not leave this as is, with the caller passing it in?